DataFormats/TrackingRecHitSoA/interface/alpaka/TrackingRecHitsSoACollection.h

```diff
@@ -34,6 +34,11 @@ namespace cms::alpakatools {
     template <typename TQueue>
     static auto copyAsync(TQueue& queue, TrackingRecHitDevice<TrackerTraits, TDevice> const& deviceData) {
       TrackingRecHitHost<TrackerTraits> hostData(queue, deviceData.view().metadata().size());
+
+      // Don't bother if zero hits
+      if (deviceData.view().metadata().size() == 0)
```

Comment:
In this https://github.com/cms-sw/cmssw/issues/45834#issuecomment-2317673908 it was suggested:

> One possibility to be foolproof would be to zero the memory in `hostData` in this case (with `std::memset()`).

I tried to do that via:

```C++
      // Don't bother if zero hits
      if (deviceData.view().metadata().size() == 0) {
        std::memset(hostData.buffer().data(),
                    0,
                    alpaka::getExtentProduct(hostData.buffer()) *
                        sizeof(alpaka::Elem<typename TrackingRecHitHost<TrackerTraits>::Buffer>));
        return hostData;
      }
      ```
      
I am not sure if that fulfills the requested suggestion and / or if it is logically correct.

Reply:
LGTM.

Note that `sizeof(...)` should always be `1`, because the buffer's element type should always be `std::byte`.

Reply:
updated in the last push.